src/smolagents/memory.py

```diff
@@ -106,7 +106,7 @@ def to_messages(self, summary_mode: bool = False, show_model_input_messages: boo
                     content=[
                         {
                             "type": "text",
-                            "text": f"Call id: {self.tool_calls[0].id}\nObservation:\n{self.observations}",
+                            "text": f"f{'Call id: {self.tool_calls[0].id}\n' if self.tool_calls else ''}Observation:\n{self.observations}",
```

Comment:
I think this would be more readable:
```suggestion
                            "text": f"Call id: {self.tool_calls[0].id}\n" if self.tool_calls else "" + f"Observation:\n{self.observations}",
```